imagetool/src/main/java/com/oracle/weblogic/imagetool/util/ARUUtil.java

```diff
@@ -45,26 +46,97 @@ private ARUUtil() {
      */
     public static String getLatestPSUNumber(FmwInstallerType category, String version, String userId, String password)
         throws Exception {
+        return getLatestPSUNumber(new SearchHelper(category, version, userId, password));
+    }
 
-        logger.entering(category, version, userId);
+    /**
+     * Get the Latest PSU patch number for the search information in the helper.
+     *
+     * @param searchHelper containing category, version, credentials for the search
+     * @return PSU patch number
+     * @throws Exception if the search or xml parse of result fails
+     */
+    public static String getLatestPSUNumber(SearchHelper searchHelper) throws Exception {
+        logger.entering(searchHelper);
         try {
             logger.info("IMG-0019");
-            String releaseNumber = getReleaseNumber(category, version, userId, password);
-            SearchResult searchResult = getRecommendedPsuMetadata(category, releaseNumber, userId, password);
-            if (searchResult.isSuccess()) {
-                Document results = searchResult.getResults();
+            searchHelper.setRelease(getReleaseNumber(searchHelper));
+            getRecommendedPsuMetadata(searchHelper);
+            if (searchHelper.isSuccess()) {
+                Document results = searchHelper.getResults();
                 String result = XPathUtil.applyXPathReturnString(results, "/results/patch[1]/name");
                 logger.exiting(result);
                 logger.info("IMG-0020", result);
                 return result;
-            } else if (!Utils.isEmptyString(searchResult.getErrorMessage())) {
-                logger.warning("IMG-0023", category, version);
-                logger.fine(searchResult.getErrorMessage());
+            } else if (!Utils.isEmptyString(searchHelper.getErrorMessage())) {
+                logger.warning("IMG-0023", searchHelper.getCategory(), searchHelper.getVersion());
+                logger.fine(searchHelper.getErrorMessage());
+            } else {
+                throw new Exception(Utils.getMessage("IMG-0032", 
+                    searchHelper.getCategory(), searchHelper.getVersion()));
+            }
+        } catch (IOException | XPathExpressionException e) {
+            throw new Exception(Utils.getMessage("IMG-0032", 
+                searchHelper.getCategory(), searchHelper.getVersion()), e);
+        }
+        logger.exiting();
+        return null;
+    }
+
+    /**
+     * Get list of PSU available for given category and release.
+     *
+     * @param category wls or fmw
+     * @param version  version number like 12.2.1.3.0
+     * @param userId   user
+     * @return Document listing of all patches (full details)
+     */
+    public static List<String> getLatestPSURecommendedPatches(FmwInstallerType category, String version,
```

Comment:
try to use camel case, and not use multiple capital letters like "PSU" back to back.

Reply:
changed PSU where used to Psu